pkg/gatewayserver/io/mock/server.go

```diff
@@ -75,6 +75,11 @@ func (s *server) FillGatewayContext(ctx context.Context, ids *ttnpb.GatewayIdent
 	return ctx, ids, nil
 }
 
+var errNoIPAddressInConnection = errors.DefineAborted(
+	"no_ip_address_in_connection",
+	"no IP address found in connection",
+)
```

Comment:
I don't know if we need to define a user error for this. If address information is missing from the context, that's an implementation bug, and we're allowed to panic.